modules/fedimint-ln-client/src/pay.rs

```diff
@@ -0,0 +1,548 @@
+        txid: TransactionId,
+    ) {
+        global_context
+            .await_tx_rejected(common.operation_id, txid)
+            .await;
+    }
+
+    async fn transition_canceled(old_state: LightningPayStateMachine) -> LightningPayStateMachine {
+        LightningPayStateMachine {
+            common: old_state.common,
+            state: LightningPayStates::Canceled,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Decodable, Encodable)]
+pub struct LightningPayFunded {
+    payload: PayInvoicePayload,
+    gateway: LightningGateway,
+}
+
+#[derive(Error, Debug, Serialize, Deserialize)]
+enum GatewayPayError {
+    #[error("Lightning Gateway failed to pay invoice")]
+    GatewayInternalError,
+}
+
+impl LightningPayFunded {
+    fn transitions(&self) -> Vec<StateTransition<LightningPayStateMachine>> {
+        let gateway = self.gateway.clone();
+        let payload = self.payload.clone();
+        let contract_id = self.payload.contract_id.clone();
+        vec![
+            StateTransition::new(
+                // Wait for the result of the payment by the gateway
+                Self::await_outgoing_contract_execution(gateway.clone(), payload.clone()),
+                move |_dbtx, result, old_state| {
+                    Box::pin(Self::transition_outgoing_contract_execution(
+                        old_state,
+                        result,
+                        contract_id,
+                    ))
+                },
+            ),
+            // wait for gatewayd for two minutes before timing out
+            StateTransition::new(
+                sleep(Duration::from_secs(120)),
+                move |_dbtx, (), old_state| {
+                    Box::pin(Self::transition_pay_timeout(old_state, contract_id))
+                },
+            ),
+        ]
+    }
+
+    async fn await_outgoing_contract_execution(
+        gateway: LightningGateway,
+        payload: PayInvoicePayload,
+    ) -> Result<String, GatewayPayError> {
+        let response = reqwest::Client::new()
+            .post(
+                gateway
+                    .api
+                    .join("pay_invoice")
+                    .expect("'pay_invoice' contains no invalid characters for a URL")
+                    .as_str(),
+            )
+            .json(&payload)
+            .send()
+            .await
+            .map_err(|_| GatewayPayError::GatewayInternalError)?;
+
+        if !response.status().is_success() {
+            return Err(GatewayPayError::GatewayInternalError);
+        }
+
+        let preimage = response
+            .text()
+            .await
+            .map_err(|_| GatewayPayError::GatewayInternalError)?;
+        let length = preimage.len();
+        Ok(preimage[1..length - 1].to_string())
+    }
+
+    async fn transition_pay_timeout(
+        old_state: LightningPayStateMachine,
+        contract_id: ContractId,
+    ) -> LightningPayStateMachine {
+        // TODO: Retry contacting gateway
```

Comment:
I didn't implement this yet. What do we want the policy to be here? Try 5 times then go to `Refundable`? Try forever until we contact the gateway?